connect/src/wormholeTransfer.ts

```diff
@@ -41,64 +41,79 @@ export enum TransferState {
   protocol: PN;
   from: SC;
   to: DC;
-  state: TransferState;
   request: TransferRequest<PN>;
+  state: TransferState;
 }
 
 export interface SourceInitiatedTransferReceipt<
   PN extends ProtocolName,
   SC extends Chain,
   DC extends Chain,
 > extends BaseTransferReceipt<PN, SC, DC> {
+  state: TransferState.SourceInitiated;
   originTxs: TransactionId<SC>[];
 }
 export interface SourceFinalizedTransferReceipt<
   PN extends ProtocolName,
   SC extends Chain,
   DC extends Chain,
-> extends SourceInitiatedTransferReceipt<PN, SC, DC> {
+> extends BaseTransferReceipt<PN, SC, DC> {
+  state: TransferState.SourceFinalized;
+  originTxs: TransactionId<SC>[];
   attestation: AttestationReceipt<PN>;
 }
 export interface AttestedTransferReceipt<
   PN extends ProtocolName,
   SC extends Chain,
   DC extends Chain,
-> extends SourceFinalizedTransferReceipt<PN, SC, DC> {
+> extends BaseTransferReceipt<PN, SC, DC> {
+  state: TransferState.Attested;
+  originTxs: TransactionId<SC>[];
   attestation: Required<AttestationReceipt<PN>>;
 }
 export interface CompletedTransferReceipt<
   PN extends ProtocolName,
   SC extends Chain,
   DC extends Chain,
-> extends AttestedTransferReceipt<PN, SC, DC> {
+> extends BaseTransferReceipt<PN, SC, DC> {
+  state: TransferState.DestinationInitiated | TransferState.DestinationFinalized;
+  originTxs: TransactionId<SC>[];
+  attestation: Required<AttestationReceipt<PN>>;
   destinationTxs: TransactionId<DC>[];
 }
 
-export function hasReachedState<PN extends ProtocolName, TS extends TransferState>(
-  receipt: TransferReceipt<PN, Chain, Chain, TransferState>,
-  state: TS,
-): receipt is TransferReceipt<PN, Chain, Chain, TS> {
-  return receipt.state === state;
+export function isAttested<PN extends ProtocolName>(
+  receipt: TransferReceipt<PN, Chain, Chain>,
+): receipt is AttestedTransferReceipt<PN, Chain, Chain> {
+  return receipt.state === TransferState.Attested;
+}
+
+export function isSourceInitiated<PN extends ProtocolName>(
+  receipt: TransferReceipt<PN, Chain, Chain>,
+): receipt is SourceInitiatedTransferReceipt<PN, Chain, Chain> {
+  return receipt.state === TransferState.SourceInitiated;
+}
+
+export function isSourceFinalized<PN extends ProtocolName>(
+  receipt: TransferReceipt<PN, Chain, Chain>,
+): receipt is SourceFinalizedTransferReceipt<PN, Chain, Chain> {
+  return receipt.state === TransferState.SourceFinalized;
 }
 
 export type TransferReceipt<
   PN extends ProtocolName,
   SC extends Chain = Chain,
   DC extends Chain = Chain,
-  TS extends TransferState = TransferState,
-> = TS extends TransferState.DestinationInitiated
-  ? CompletedTransferReceipt<PN, SC, DC>
-  : TS extends TransferState.Attested
-  ? AttestedTransferReceipt<PN, SC, DC>
-  : TS extends TransferState.SourceFinalized
-  ? SourceFinalizedTransferReceipt<PN, SC, DC>
-  : TS extends TransferState.SourceInitiated
-  ? SourceInitiatedTransferReceipt<PN, SC, DC>
-  : never;
+> =
+  | BaseTransferReceipt<PN, SC, DC>
```

Comment:
oh maybe needs to be exported because of this?

Reply:
oh right. I could try to get rid of this